docs/source/html_builder.py

```diff
@@ -11,16 +11,35 @@
 from sphinx.builders.html import StandaloneHTMLBuilder
 from sphinx.environment import BuildEnvironment
 
+from mypy.main import process_options
+
 
 class MypyHTMLBuilder(StandaloneHTMLBuilder):
+    strict_file: Path
+
     def __init__(self, app: Sphinx, env: BuildEnvironment) -> None:
         super().__init__(app, env)
         self._ref_to_doc = {}
+        self.strict_file = Path(self.srcdir) / "strict_list.rst"
+        self._add_strict_list()
 
     def write_doc(self, docname: str, doctree: document) -> None:
         super().write_doc(docname, doctree)
         self._ref_to_doc.update({_id: docname for _id in doctree.ids})
 
+    def _add_strict_list(self) -> None:
+        strict_flags: list[str] = []
+        process_options(["-c", "pass"], list_to_fill_with_strict_flags=strict_flags)
```

Comment:
Suggestion: instead of adding an ad-hoc argument, consider refactoring parser builder logic away from `process_options` to a separate function, returning `(parser, strict_flag_names)` as a tuple. Then `process_options` will get shorter (which is quite great for a 1000+ lines function) and you won't need a flag, you'll only build a parser and discard it immediately.

I didn't check if there are more dependencies, sorry if so

Reply:
I was pretty timid about refactoring process_options, but on reflection I think this is a good suggestion.

list_to_fill_with_strict_flags is my invention, so nothing else needs it.